packages/components/src/Notification/subcomponents/GenericNotification/GenericNotification.tsx

```diff
@@ -49,89 +56,102 @@ export type GenericNotificationVariant = {
 export type GenericNotificationProps = GenericNotificationBase &
   (GenericNotificationType | GenericNotificationVariant)
 
-export const GenericNotification = ({
-  type,
-  variant,
-  style,
-  children,
-  title,
-  persistent = false,
-  onHide,
-  noBottomMargin,
-  forceMultiline,
-  headingProps,
-  classNameOverride,
-  ...restProps
-}: GenericNotificationProps): JSX.Element | null => {
-  const [isHidden, setIsHidden] = useState<boolean>(true)
-  const [isRemoved, setIsRemoved] = useState<boolean>(false)
+export const GenericNotification = forwardRef<
+  HTMLDivElement,
+  GenericNotificationProps
+>(
+  (
+    {
+      type,
+      variant,
+      style,
+      children,
+      title,
+      persistent = false,
+      onHide,
+      noBottomMargin,
+      forceMultiline,
+      headingProps,
+      classNameOverride,
+      ...restProps
+    },
+    ref
+  ): JSX.Element | null => {
+    const [isHidden, setIsHidden] = useState<boolean>(true)
+    const [isRemoved, setIsRemoved] = useState<boolean>(false)
 
-  const containerRef = useRef<HTMLDivElement>(null)
+    const containerRef = isRefObject(ref) ? ref : useRef<HTMLDivElement>(null)
```

Comment:
Something like this since hooks shouldn't be called conditionally.

```suggestion
    const fallbackRef = useRef<HTMLDivElement>(null)
    const containerRef = isRefObject(ref) ? ref : fallbackRef
```